src/EnergyPlus/DataHeatBalance.cc

```diff
@@ -1309,52 +1308,58 @@ Real64 ComputeNominalUwithConvCoeffs(EnergyPlusData &state,
     Real64 outsideFilm;
 
     isValid = true;
+
+    auto &thisSurface = state.dataSurface->Surface(numSurf);
+
     // exterior conditions
-    switch (state.dataSurface->Surface(numSurf).ExtBoundCond) {
-    case ExternalEnvironment: {
-        outsideFilm = 0.0299387; // All exterior conditions
+    switch (thisSurface.ExtBoundCond) {
+    case DataSurfaces::ExternalEnvironment: { // ExtBoundCond = 0
+        outsideFilm = 0.0299387;              // All exterior conditions
     } break;
-    case Ground:
-    case GroundFCfactorMethod: {
-        outsideFilm = 0.0; // No outside film when underground
+    case DataSurfaces::Ground:
+    case DataSurfaces::OtherSideCoefNoCalcExt:
+    case DataSurfaces::OtherSideCoefCalcExt:
+    case DataSurfaces::OtherSideCondModeledExt:
+    case DataSurfaces::GroundFCfactorMethod:
+    case DataSurfaces::KivaFoundation: { // All these cases have a negative ExtBoundCond so don't use film coefficients
+        outsideFilm = 0.0;
     } break;
-    default: {
-        if (state.dataSurface->Surface(numSurf).ExtBoundCond > 0) { // interzone partition
-            // use companion surface in adjacent zone
-            switch (state.dataSurface->Surface(state.dataSurface->Surface(numSurf).ExtBoundCond).Class) {
-            case SurfaceClass::Wall:
-            case SurfaceClass::Door: { // Interior:  vertical, still air, Rcin = 0.68 ft2-F-hr/BTU
-                outsideFilm = 0.1197548;
-            } break;
-            case SurfaceClass::Floor: { // Interior:  horizontal, still air, heat flow downward, Rcin = 0.92 ft2-F-hr/BTU
-                outsideFilm = 0.1620212;
-            } break;
-            case SurfaceClass::Roof: { // Interior:  horizontal, still air, heat flow upward, Rcin = 0.61 ft2-F-hr/BTU
-                outsideFilm = 0.1074271;
-            } break;
-            default: {
-                outsideFilm = 0.0810106; // All semi-exterior surfaces
-            } break;
-            }
-        } else {
-            outsideFilm = 0.0810106; // All semi-exterior surfaces
+    default: { // Interior Surface Attached to a Zone (ExtBoundCond is a surface)
+        switch (state.dataSurface->Surface(thisSurface.ExtBoundCond).Class) {
```

Comment:
If you are only trying to assign a constant value based on an enumeration, a smaller/faster idiom is:

```cpp
static constexpr std::array<Real64, static_cast<int>DataSurfaces::SurfaceClass::Num> outsideFilmCoefs = { 
   0.1197584, // Wall
   0.1620212, // Floor
   0.1074271, // Roof 
   ... 
};

outsideFilm = outsideFilmCoefs[static_cast<int>(state.dataSurface->Surface(thisSurface.ExtBoundCond).Class)];
```

Reply:
That converts an entire switch/case into a single lookup.

Reply:
I have pushed a new commit that changes this as you suggested.

As for Invalid vs. None, I don't know that "None" has any specific reason to be there.  A search of SurfaceClass::None in the code only results in four occurrences.  I'll see if I can get rid of None without too much trouble and make another commit if I don't run into any complications.

Reply:
Ok, I've tried this and while None is not really used for anything the integer evaluation of SurfaceClass is used as indices for other arrays.  It skips the "None" which is evaluating to 0.  Getting rid of None means that Wall becomes zero which leads to index errors.  So, it looks like there is a "use" for None though maybe at some point things could be rewritten to get rid of that case.  For now, I plan on leaving None in and move forward with the other changes unless someone has an issue with that.

Reply:
This is an artefact of Fortran 1-based arrays and the use of 0 to mean "not found" or "invalid". In the C/C++ world, 0 is valid of course and -1 is used for "invalid" or "not found". 

It may be worth taking a quick stab to see if the arrays indexed by `SurfaceClass` (and which are causing the index errors) can be switched over from `Array1D` to `std::array` without requiring knock-on changes.  If so, great. If not, 📌 . If too tired, also 📌 .

Reply:
Let's 📌 this for now.  I'm wondering if at some point someone will need to look beyond just this one and find other items like this.  Hopefully the other changes here are solid and ci continues to come back clean.